tests/run-make/print-target-list/rmake.rs

```diff
@@ -0,0 +1,15 @@
+// Checks that all the targets returned by `rustc --print target-list` are valid
+// target specifications
+
+//@ needs-llvm-components: aarch64 arm avr bpf csky hexagon i386 i586 i686 loongarch64
+//@ needs-llvm-components: m68k mips msp430 nvptx64 powerpc riscv s390x sparc thumb wasm x86 xtensa
+
+use run_make_support::bare_rustc;
+
+fn main() {
+    let targets = bare_rustc().print("target-list").run().stdout_utf8();
+
+    for target in targets.lines() {
+        bare_rustc().target(target).print("sysroot").run();
```

Comment:
Suggestion: we should skip the check (and leave a FIXME here) if the target is experimental, i.e.

https://github.com/rust-lang/rust/blob/1df0458781d6fd753a68c4cdc4de5313b1635dbd/src/bootstrap/src/core/build_steps/llvm.rs#L325-L328

skip if one of AVR, M68k, CSKY, Xtensa.